### PR TITLE
Follow up - Engage upgrade

### PR DESCRIPTION
**Pull request by sc-addypathania**
This is a follow up of Engage upgrade [1609](https://github.com/Sitecore/jss/pull/1609)
- Upgrade engage to 1.4.1
- Refactor middleware to fix sub-domain cookie bug.

- [x] This PR follows the [Contribution Guide](https://github.com/Sitecore/jss/blob/dev/CONTRIBUTING.md)
- [ ] Changelog updated

## Testing Details

- [ ] Unit Test Added
- [x] Manual Test/Other (Please elaborate)

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: packages/sitecore-jss-nextjs/src/middleware/personalize-middleware.ts [hunk 1]
```diff
@@ -240,18 +245,24 @@ export class PersonalizeMiddleware extends MiddlewareBase {
...
     response.headers.set('x-middleware-cache', 'no-cache');
-    // Share rewrite path with following executed middlewares
+    // Share rewrite path with following executed middleware
     response.headers.set('x-sc-rewrite', rewritePath);
-
     // Share site name with the following executed middlewares
     response.cookies.set(this.SITE_SYMBOL, site.name);
 
+    if (setCookieHeader) {
+      response.headers.set('set-cookie', setCookieHeader);
```
Comment (ambrauer) [inline packages/sitecore-jss-nextjs/src/middleware/personalize-middleware.ts:263]: This looks a bit different than I was expecting. Wouldn't this override the preceding `response.cookies.set(this.SITE_SYMBOL, site.name);`? 

Might be worth explaining what's going on here with the get/set of 'set-cookie' header (and why we can't use `cookies.set`).

Comment (sc-addypathania) [reply packages/sitecore-jss-nextjs/src/middleware/personalize-middleware.ts]: makes sense.
Refactored.

File: packages/sitecore-jss-nextjs/src/middleware/personalize-middleware.ts [hunk 2]
```diff
@@ -240,18 +245,33 @@ export class PersonalizeMiddleware extends MiddlewareBase {
...
+    // browserId cookie gets omitted after rewrite
+    cookies.forEach((cookie) => {
+      response.cookies.set(cookie.name, cookie.value, {
+        domain: cookie.domain,
+        expires: cookie.expires,
+        maxAge: cookie.maxAge,
+        sameSite: cookie.sameSite,
+        secure: cookie.secure,
+      });
+    });
```
[the remark this answers is not in the document]
Comment (sc-addypathania) [reply packages/sitecore-jss-nextjs/src/middleware/personalize-middleware.ts:273]: It does not handle the optional cookie attributes.

[the remark this answers is not in the document]
Comment (ambrauer) [reply packages/sitecore-jss-nextjs/src/middleware/personalize-middleware.ts:273]: That's a bummer. Alright, approving :)

**Review by ambrauer — APPROVED** [review]

File: packages/sitecore-jss-nextjs/src/middleware/personalize-middleware.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (sc-addypathania) [reply packages/sitecore-jss-nextjs/src/middleware/personalize-middleware.ts]: Sorry my bad, this works fine. 😪

**Pull request merged**